src/Raven.Server/Documents/DocumentPutAction.cs

```diff
@@ -240,6 +240,15 @@ public PutOperationResults PutDocument(DocumentsOperationContext context, string
                     }
                 }
 
+                if (document.TryGetMetadata(out BlittableJsonReaderObject docMetadata))
+                {
+                    if (oldValue.Pointer == null)
```

Comment:
we want that only for new documents? what if this is an update, how it will work?

Reply:
I think that we want that only for new/cloned documents. 

If just adding archived true to document would make it archived, we would need to make license check here.

Now only way to archive documents is by having DataArchivist enabled in data archival configuration. 

This would implicate that documents should be unarchived when the flag is removed within a document update, and I'm not sure if these changes are in the scope of the issue.

If we want to make this change anyway, let me know.

Reply:
In term of storage, it adds documentFlag to the doc. Indexes have [Archived Behavior](https://ravendb.net/docs/article-page/7.1/csharp/server/extensions/archival#section), archived documents are skipped by default. The same case with subscriptions.

Reply:
So this is pretty much just a flag, no?
So if I want to _manually_ do something like that, why should I be prevented from doing this?

Reply:
It's a licensed feature - we'd need to make License check in `PutDocument`

Reply:
Currently only way to use the feature is setting up Archival Configuration (similar to Expiration).

Reply:
No, don't do that. If the user _explicitly_ set the `@archived: true `, etc - let it run, no need to do anything.
The archival licensing aspect is the _moving_ to achive mode

Reply:
Not sure that I follow the logic.
With this table - I can _only_ move to archive mode, but never out? 
And the only way to do that is via the archiver directly?

Reply:
By design you _unarchive_ your documents with the Patch - `archived.unarchive(doc)` in the patch script
This PR introduces nonPersistentFlag, which marks this change to a document, so it's not confused.